client-app/shared/layout/components/header/_internal/mega-menu.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <nav :id="componentId" ref="megaMenuElement" class="mega-menu">
+    <VcPopover arrow-enabled placement="bottom-start" class="mega-menu__popover">
+      <template #trigger>
+        <button type="button" to="/catalog" class="mega-menu__button" :disabled="loading">
+          <VcLoader v-if="loading" class="mega-menu__loader" />
+          <VcIcon v-else class="mega-menu__icon" name="drag-dots" />
+
+          <span> {{ $t("common.buttons.all_products") }} </span>
+        </button>
+      </template>
+
+      <template #content="{ close }">
+        <div class="mega-menu__content">
+          <Subcategories v-if="category" :item="category" @close="close" />
+        </div>
+      </template>
+    </VcPopover>
+
+    <div class="mega-menu__divider"></div>
+
+    <ul ref="navElement" class="mega-menu__nav">
+      <li v-for="(item, index) in visibleItems" :key="index" class="mega-menu__item" menu-item>
+        <router-link :to="item.route ?? '#'" class="mega-menu__link">
+          {{ item.title }}
+        </router-link>
+      </li>
+    </ul>
+  </nav>
+</template>
+
+<script setup lang="ts">
+import { useElementBounding, watchDebounced } from "@vueuse/core";
+import { ref, computed, onMounted, useTemplateRef, nextTick } from "vue";
+import { useNavigations } from "@/core/composables";
+import { useCategory } from "@/shared/catalog";
+import { useComponentId } from "@/ui-kit/composables";
+import Subcategories from "./subcategories.vue";
+
+interface IProps {}
+
+defineProps<IProps>();
+
+const calculating = ref(false);
+const navRef = useTemplateRef<HTMLElement>("navElement");
+const menuRef = useTemplateRef<HTMLElement>("megaMenuElement");
+const visibleItemsCount = ref(1);
+
+const { width: menuRight } = useElementBounding(menuRef);
+
+const componentId = useComponentId("mega-menu");
```

Comment:
Why you use this componentId? 

If for E2E - it is not recommended because getCurrentInstance().uid unstable.